### PR TITLE
Add handlePR unit tests for trigger plugin

### PR DESCRIPTION
**Pull request by aojea**
Add `handlePR` unit tests for trigger plugin

Ref: #8602

### CODE REVIEW

File: prow/plugins/trigger/pr_test.go [hunk 1]
```diff
@@ -147,3 +150,215 @@ func TestTrusted(t *testing.T) {
...
+		Author        string
+		ShouldBuild   bool
+		ShouldComment bool
+		HasOkToTest   bool
+		prLabel       string
+		prChanges     bool
+		prAction      github.PullRequestEventAction
+	}{
+		{
+			name: "Trusted user Open PR.",
```
Comment (fejta) [inline prow/plugins/trigger/pr_test.go]: Ideally this says what is supposed to happen, not just the scenario.

AKA -- `Trusted user open PR should build`

File: prow/plugins/trigger/pr_test.go [hunk 2]
```diff
@@ -147,3 +150,215 @@ func TestTrusted(t *testing.T) {
...
+		},
+		{
+			name: "Trusted user Edit PR without changes",
+
+			Author:      "t",
+			ShouldBuild: false,
+			prAction:    github.PullRequestActionEdited,
+		},
+		{
+			name: "Untrusted user Edit PR.",
```
Comment (fejta) [inline prow/plugins/trigger/pr_test.go]: This should build if there's an ok-to-test (same for sync)

**Comment by fejta** [comment]
Thanks!

**Comment by aojea** [comment]
@fejta I updated the PR with your comments

**Pull request merged**